src/coreclr/jit/gentree.cpp

```diff
@@ -11728,41 +11728,53 @@ GenTree* Compiler::gtFoldHWIntrinsic(GenTreeHWIntrinsic* node)
 
     switch (node->GetHWIntrinsicId())
     {
-#if defined(TARGET_XARCH) || defined(TARGET_ARM64)
         case NI_Vector128_Create:
-        {
-            if ((node->GetOperandCount() == 1) &&
-                (node->Op(1)->IsIntegralConst(0) || node->Op(1)->IsFloatPositiveZero()))
-            {
-                node->ResetHWIntrinsicId(NI_Vector128_get_Zero);
-            }
-            break;
-        }
-#endif
-
 #if defined(TARGET_XARCH)
         case NI_Vector256_Create:
+#elif defined(TARGET_ARMARCH)
+        case NI_Vector64_Create:
+#endif
         {
-            if ((node->GetOperandCount() == 1) &&
-                (node->Op(1)->IsIntegralConst(0) || node->Op(1)->IsFloatPositiveZero()))
+            bool hwAllArgsAreConstZero = true;
+            for (GenTree** use : node->UseEdges())
             {
-                node->ResetHWIntrinsicId(NI_Vector256_get_Zero);
+                if (!(*use)->IsIntegralConst(0) && !(*use)->IsFloatPositiveZero())
+                {
+                    hwAllArgsAreConstZero = false;
+                    break;
+                }
```

Comment:
I know this is WIP, but there is an `Operands` iterator that's intended to be used where the node doesn't need to replaced.

Reply:
The current iteration technique is already used in `fgMorphMultiOp`, I just copied from it.
I will look at the `Operands` iterator though.

Reply:
Yea, using the `Operands` iterator is a lot cleaner. :) Thank you!